hasktorch/src/Torch/NN.hs

```diff
@@ -118,12 +119,12 @@ class Parameterized f where
   default _replaceParameters :: (Generic f, GParameterized (Rep f)) => f -> ParamStream f
   _replaceParameters f = to <$> _gReplaceParameters (from f)
 
-  replaceDevice :: Device -> f -> f
-  default replaceDevice :: (Generic f, GParameterized (Rep f)) => Device -> f -> f
-  replaceDevice dev f = to $ gReplaceDevice dev (from f)
+  replaceTensor :: (Tensor -> Tensor) -> f -> f
```

Comment:
this is a nice idea. can you make this dependent on the generic representation if `f`, please?
this way, one could use that to do more intentional model weight surgery.

Reply:
I think there is a demand to match and replace specific data, but I'm not sure how to make it without creating a custom class and instance for each application. I wish there was something like try-function for to-function of ghc.generic.

Reply:
We can do it with template haskell, but it's not what we expect.

Reply:
Oh, I misunderstood it.

Reply:
I misunderstood a little, please wait a moment.